sdk/core/azure-core/tests/test_universal_pipeline.py

```diff
@@ -408,3 +414,68 @@ def emit(self, record):
     assert mock_handler.messages[4].message == 'Response headers:'
 
     mock_handler.reset()
+
+def test_retry_seekable_stream():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self._first = True
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            if self._first:
+                self._first = False
+                request.body.seek(0,2)
+                raise AzureError('fail on first')
+            position = request.body.tell()
+            assert not position
+            return HttpResponse(request, None)
+
+    data = BytesIO(b"Lots of dataaaa")
+    http_request = HttpRequest('GET', 'http://127.0.0.1/')
+    http_request.set_streamed_data_body(data)
+    http_retry = RetryPolicy(retry_total = 1)
+    pipeline = Pipeline(MockTransport(), [http_retry])
+    pipeline.run(http_request)
+
+def test_retry_seekable_file():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self._first = True
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            if self._first:
+                self._first = False
+                for value in request.files.values():
+                    name, body = value[0], value[1]
+                    if name and body and hasattr(body, 'read'):
+                        body.seek(0,2)
+                        raise AzureError('fail on first')
+            for value in request.files.values():
+                name, body = value[0], value[1]
+                if name and body and hasattr(body, 'read'):
+                    position = body.tell()
+                    assert not position
+                    return HttpResponse(request, None)
+
+    http_request = HttpRequest('GET', 'http://127.0.0.1/')
+    headers = {'Content-Type': "multipart/form-data"}
+    http_request.headers = headers
+    form_data_content = {
+        'fileContent': open('tmpfile.txt', 'rb'),
+        'fileName': 'tmpfile',
+    }
+    http_request.set_formdata_body(form_data_content)
+    http_retry = RetryPolicy(retry_total = 1)
+    pipeline = Pipeline(MockTransport(), [http_retry])
+    pipeline.run(http_request)
```

Comment:
It is possible to compare the finally received data to the expected/sent data in both of these?

Reply:
I would prefer this test to focus on checking rewind behaviors only and not make it complicated.